cli/azd/pkg/commands/pipeline/github_provider.go

```diff
@@ -323,14 +325,19 @@ type GitHubCiProvider struct {
 	credential    azcore.TokenCredential
 	commandRunner exec.CommandRunner
 	console       input.Console
+	httpClient    httputil.HttpClient
 }
 
 func NewGitHubCiProvider(
-	credential azcore.TokenCredential, commandRunner exec.CommandRunner, console input.Console) *GitHubCiProvider {
+	credential azcore.TokenCredential,
+	commandRunner exec.CommandRunner,
+	console input.Console,
+	httpClient httputil.HttpClient) *GitHubCiProvider {
```

Comment:
I still wonder if the amount of threading here just to reshare a single client is worth it.

I do wonder if it's simpler to just have the new function here reference a static package variable, and have a separate function for substitution under test, or simply by constructing the struct directly using privates.